openapi.yaml

```diff
@@ -164,1090 +135,1589 @@ paths:
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
     post:
-      description: Create phone numbers for a person
+      description: Create phone number for a person
       tags:
         - phones
-      operationId: createPhones
+      operationId: postPhone
       parameters:
         - name: phoneBody
           in: body
-          description: Post body for the new Phone
+          description: Post body for the new phone
           required: true
           schema:
-            $ref: '#/definitions/NewPhoneResultObject'
-        - $ref: '#/parameters/osuID'
-        - $ref: '#/parameters/authorization'
+            $ref: '#/definitions/PhonePostBody'
+        - $ref: '#/parameters/osuId'
       responses:
-        200:
+        '200':
           description: Successful response
           schema:
-            $ref: '#/definitions/PhoneResultObject'
-        400:
-          description: Bad request
-          schema:
-            $ref: '#/definitions/ErrorArray'
-        404:
-          description: Person not found
+            $ref: '#/definitions/PhoneResult'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '409':
+          $ref: '#/responses/409'
+        '500':
+          $ref: '#/responses/500'
+  /persons/{osuId}/emails:
+    get:
+      description: Get emails for a person
+      tags:
+        - emails
+      operationId: getEmails
+      parameters:
+        - $ref: '#/parameters/osuId'
+        - name: filter[emailType]
+          in: query
+          type: array
+          items:
+            type: string
+            collectionFormat: csv
+          required: false
+          enum: &emailTypeEnums
+            - ONID
+            - STU
+            - VEND
+            - VWEB
+            - EMPL
+            - VPAY
+            - EMER
+          description: |
+            Type of email
+            - `ONID` - Student
+            - `STU` - Non-directory Student
+            - `VEND` - Vendor
+            - `VWEB` - Vendor
+            - `EMPL` - Employee
+            - `VPAY` - Vendor Pay
+            - `EMER` - Emergency Contact Persons Email for Student
+        - name: filter[preferredInd]
+          in: query
+          type: boolean
+          description: Filter on preferred email addresses
+      responses:
+        '200':
+          description: Successful response
           schema:
-            $ref: '#/definitions/Error'
-
+            $ref: '#/definitions/EmailSetResult'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
 parameters:
-  osuID:
-    name: osuID
+  osuId:
+    name: osuId
     in: path
     required: true
-    description: OSU ID of person
+    description: OSU ID of a person
     type: string
-  jobID:
-    name: jobID
+    pattern: '^\d{9}$'
```

Comment:
Since we have pattern for this path parameter, we should return 400 if input doesn't match the regex.

Reply:
Didn't we decide that for id path parameters, if it doesn't match the pattern, we should still respond with a 404?

Reply:
@ianbrown9475 You're right, I was looking into the wrong place, feel free to ignore my comment here.